project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/execution/exception/KotlincExceptionUtils.kt

```diff
@@ -0,0 +1,116 @@
+package org.plan.research.minimization.plugin.execution.exception
+
+import arrow.core.Either
+import arrow.core.Option
+import arrow.core.getOrElse
+import arrow.core.raise.either
+import arrow.core.raise.ensure
+import arrow.core.raise.ensureNotNull
+import arrow.core.raise.option
+import com.intellij.build.events.BuildEvent
+import com.intellij.build.events.FileMessageEvent
+import com.intellij.build.events.MessageEvent
+import org.jetbrains.kotlin.utils.addToStdlib.indexOfOrNull
+import org.plan.research.minimization.plugin.errors.KotlincExceptionError
+import org.plan.research.minimization.plugin.fromFilePosition
+import org.plan.research.minimization.plugin.fromString
+import org.plan.research.minimization.plugin.model.CaretPosition
+
+object KotlincExceptionUtils {
```

Comment:
Let's make it as a class

Reply:
Because it seems like some parsing stuff, so `class KotlincExceptionParser` would be more informative (and potentially scalable)

Reply:
FIX: not a parser, more a translator